src/Features/Core/Shared/Utilities/PatternMatcher.cs

```diff
@@ -165,9 +165,19 @@ public PatternMatcher(string pattern, CultureInfo culture, bool verbatimIdentifi
             _compareInfo = culture.CompareInfo;
 
             _fullPatternSegment = new Segment(pattern, verbatimIdentifierPrefixIsWordCharacter);
-            _dotSeparatedSegments = pattern.Split(s_dotCharacterArray, StringSplitOptions.RemoveEmptyEntries)
-                                            .Select(text => new Segment(text.Trim(), verbatimIdentifierPrefixIsWordCharacter))
-                                            .ToArray();
+
+            if (pattern.IndexOf('.') < 0)
+            {
+                // PERF: Avoid string.Split allocations when the pattern doesn't contain a dot.
+                _dotSeparatedSegments = pattern.Length > 0 ? new Segment[1] { _fullPatternSegment } : SpecializedCollections.EmptyArray<Segment>();
+            }
+            else
+            {
+                _dotSeparatedSegments = pattern.Split(s_dotCharacterArray, StringSplitOptions.RemoveEmptyEntries)
```

Comment:
Interestingly, I heard on Friday talking to some of the corefx folks that string.Split allocations have been coming up a lot lately, and things might change in the framework.  @nguerrera @ellismg